### DISCUSSION TITLE
VisuallyHidden needs prop showOnInteraction

### DISCUSSION THREAD

**Discussion by bryanrasmussen** [chakra-ui/chakra-ui#8673] [Ideas]
### Description

It should be possible to make a visuallyhidden component visible on user interaction

### Problem Statement/Justification

 it is often wanted for accessibility to make the visually hidden component visible once it is interacted with.  It is often complained by different accessibility testers if the visible page content varies too much from the available page content. 

### Proposed Solution or API

Either the VisuallyHidden component should take _focus attributes the way others do  - which it does not seem to as the following does not work:

<VisuallyHidden _focus={
                    { position: "static",
                      width: "auto",
                      height: "auto",
                      backgroundColor: "white",
                      color: "black"}
                }> 
                
  or there should be a showOnInteraction prop that when set to true removes visually hidden styling on focus event 

### Alternatives

as discussed I tried making a _focus but did not get an css class name::focus out, also tried same with sx={{_focus etc. did not get anything. 

### Additional Information

http://web-accessibility.carnegiemuseums.org/code/skip-link/
"Hide the skip link from sighted users – styling the link tag with what is typically called and classified as ‘screenreader-text’ – which only appears when focused on by tabbing to it or using other assistive technology."

obviously you do not always want visually hidden info to become visible when interacted with but often it is the what is wanted.

**Comment by segunadebayo** [comment]
`VisuallyHidden` stays hidden. Build a skip link with `_focus` on a real `Link`.

Do not change the primitive for the skip-link case.
